core/src/main/java/org/jivesoftware/sparkimpl/plugin/transcripts/ChatTranscript.java

```diff
@@ -71,20 +71,33 @@ public void addHistoryMessage(HistoryMessage entry) {
      * @return the messages that included search keywords.
      */
     public List<HistoryMessage> getMessage(String text) {
-    	if(text == null || "".equals(text)) {
-    		return messages;
-    	} else {
-	    	List<HistoryMessage> searchResult = new ArrayList<>();
-	    	for(HistoryMessage message : messages) {
-	    		// ignore keywords' case
-	    		if( message.getBody().toLowerCase().contains( text.toLowerCase() ) ) {
-	    			searchResult.add(message);
-	    		}
-	    	}
-	    	return searchResult;
-    	}
+        if (text == null || "".equals(text)) {
+            for (HistoryMessage message : messages) {
+                message.setBody(message.getBody().toLowerCase().replaceAll("#ffff00","#ffffff"));
+            }
+            return messages;
+        } else {
+            List<HistoryMessage> searchResult = new ArrayList<>();
+            int iter = 0;
+            HistoryMessage previous = null;
+            for(HistoryMessage message : messages) {
+                iter++;
+                if (message.getBody().toLowerCase().contains(text.toLowerCase())) {
+                    message.setBody(message.getBody().toLowerCase().replaceAll(text.toLowerCase(),
+                            "<span style=\"background-color: #ffff00\">" + text.toLowerCase() + "</span>"));
+
+                    if (previous != null) searchResult.add(previous);
```

Comment:
Please always wrap if-statements, even if they are just one line. Not wrapping them makes for easy mistakes.